crates/hyperqueue/src/dashboard/state.rs

```diff
@@ -1,28 +1,51 @@
 use crate::dashboard::data::DashboardData;
+use crate::dashboard::ui::screen::controller::{DashboardScreenState, ScreenController};
 use crate::dashboard::ui::screen::Screen;
-use crate::dashboard::ui::screens::home::screen::HomeScreen;
+use crate::dashboard::ui::screens::home::screen::ClusterOverviewScreen;
+use crate::dashboard::ui::screens::worker::screen::WorkerOverviewScreen;
 use tako::common::WrappedRcRefCell;
 
 pub struct DashboardState {
-    screens: Vec<Box<dyn Screen>>,
-    current_screen: usize, // TODO: usize newtype wrapper or enum
+    cluster_overview_screen: ClusterOverviewScreen,
+    worker_overview_screen: WorkerOverviewScreen,
+
     data_source: WrappedRcRefCell<DashboardData>,
+
+    current_screen: DashboardScreenState,
```

Comment:
Let's create the `DashboardScreenState` enum here. It should be a simple enum with fieldless variants, it simply says what screen is active, without holding any other state. It's something different than a command to change a screen (even though it looks similar).

We could also represent the screen state by a vector of screens and an index that would remember which state is active. But for now this design is OK.